### DISCUSSION TITLE
Cannot generate allure report - child process error

### DISCUSSION THREAD

**Discussion by tsbnode** [allure-framework/allure2#2519] [Bug Reports] [answered]
Cannot generate allure report - child process error
>allure serve

node:internal/child_process:421
    throw new ErrnoException(err, 'spawn');
    ^

Error: spawn EINVAL
    at ChildProcess.spawn (node:internal/child_process:421:11)
    at Object.spawn (node:child_process:761:9)
    at module.exports (C:\qa\wdio-tests\node_modules\allure-commandline\index.js:7:37)
    at Object.<anonymous> (C:\qa\wdio-tests\node_modules\allure-commandline\bin\allure:5:1)
    at Module._compile (node:internal/modules/cjs/loader:1369:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1427:10)
    at Module.load (node:internal/modules/cjs/loader:1206:32)
    at Module._load (node:internal/modules/cjs/loader:1022:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49 {
  errno: -4071,
  code: 'EINVAL',
  syscall: 'spawn'
}

Node.js v20.12.2,

package.json values:
allure-commandline: "^2.28.0",
@wdio/allure-reporter: "^7.33.0",
@wdio/cli": "^7.7.4",
@wdio/cucumber-framework": "7.34.0",

**Comment by baev** [comment]
see https://github.com/allure-framework/allure-npm/issues/43

We successfully [reproduced the issue](https://github.com/allure-framework/allure-npm/pull/44), will provide a fix ASAP

**Reply by baev** [reply, marked as answer]
Will be released as 2.29.0 within a few hours (the release is in process already)

**Comment by Tamil3034** [comment]
Getting same error while using  "**allure-commandline": "^2.29.0**"

**Reply by baev** [reply]
Could you please specify the node version?

**Reply by Tamil3034** [reply]
v21.7.3

**Reply by vibin-c** [reply]
I tried with v21.7.3 and with v22.0.0 also, it got only worked when i installed previous version of node v20.10.0

**Reply by baev** [reply]
I don't know how we can help you. The issue isn't with Allure. It was a breaking change released by Node due to [a high-security vulnerability](https://nodejs.org/en/blog/vulnerability/april-2024-security-releases-2). 

The change is that now, to use `spawn` and `spawnAsync` to run `.cmd` or `.bat` files, you need to specifically provide the`{ shell: true}` option; otherwise, Node will throw the `EINVAL` exception.

Since `allure-commandline@2.29.0`, we added the [required option](https://github.com/allure-framework/allure-npm/blob/main/index.js#L10), [the fix was tested using Node 18, 20 and 21](https://github.com/allure-framework/allure-npm/actions/runs/8816925361).

We can't reproduce the issue anymore on our side. The only recommendation is to double-check the Node version; it should be the latest available minor/patch for each major node version.

BTW the current LTS is `v20.12.2` till the end of the year, so why don't just use it for now?

**Reply by vibin-c** [reply]
Update: Issue resolved, Thanks for the support. I reinstalled with latest version 22.0.0 of node and allure 2.29.0 and restarted the system it worked.

**Reply by Tamil3034** [reply]
While using v20.12.2 also getting the same error
![image](https://github.com/allure-framework/allure2/assets/167416857/ad3a86b9-fe76-4e04-b9c7-b5745c8a2bc8)

**Comment by antonyGusev** [comment]
I've getting the same error with node 22.6.0 and allure-commandline 2.30.0
